File: EEntityCore.DB/EEntityCore.DB.MSSQL/Exceptions/SQLServerConnectionException.vb
Imports CODERiT.Logger.Standard.VB.Exceptions

Namespace Exceptions


    REM Encrypt connection string

    Public Class SQLServerConnectionException
        Inherits EException

#Region "Constructors"

        Public Sub New(pValConnectionString As String)

            Me.New("INVALID CONNECTION", pValConnectionString)

        End Sub


        Sub New(ByVal ex As Exception, pValConnectionString As String)

            Me.New(ex.Message, pValConnectionString, ex)

        End Sub



        Sub New(ByVal Title As String, pValConnectionString As String, ByVal ex As Exception)


            Me.Load(Title,
                    New String() {"Check to see that the connection parameters are correct",
                                  "Check to see if this pc can ping the server ip.",
                              "CONNECTION STRING: " & pValConnectionString},
                    New String() {ex.Message, ex.StackTrace}
                    )


        End Sub



        Sub New(ByVal Title As String, pValConnectionString As String)

            Me.New(Title, pValConnectionString, New Exception(Title))

        End Sub




#End Region



    End Class


End Namespace

File: EEntityCore.DB/EEntityCore.DB.MSSQL/Exceptions/SQLServerConnectionException.cs
using System;
using CODERiT.Logger.Standard.VB.Exceptions;

namespace EEntityCore.DB.MSSQL.Exceptions
{


    // REM Encrypt connection string

    public class SQLServerConnectionException : EException
    {

        #region Constructors

        public SQLServerConnectionException(string pValConnectionString) : this("INVALID CONNECTION", pValConnectionString)
        {
        }

        public SQLServerConnectionException(Exception ex, string pValConnectionString) : this(ex.Message, pValConnectionString, ex)
        {
        }

        public SQLServerConnectionException(string Title, string pValConnectionString, Exception ex)
        {
            Load(Title, new string[] { "Check to see that the connection parameters are correct", "Check to see if this pc can ping the server ip.", "CONNECTION STRING: " + pValConnectionString }, new string[] { ex.Message, ex.StackTrace });
        }

        public SQLServerConnectionException(string Title, string pValConnectionString) : this(Title, pValConnectionString, new Exception(Title))
        {
        }




        #endregion



    }
}